### PR TITLE
Display issuer on Configuration details

### PR DESCRIPTION
**Pull request by Monkeychip**
### Description
* For enterprise users who have modified WIF root-config attributes, show `Issuer` on the Configuration details view. 
* For this flow, I ended up creating a model and adapter called `identity/oidc/config` to handle the query and create/edit UX. It's likely we'll use this same request in Azure and GCP configs. T[his is the API doc](https://developer.hashicorp.com/vault/api-docs/secret/identity/tokens#configure-the-identity-tokens-backend) for this model/adapter. Thank you Claire for help on naming here.

Welp—it was more changes than I anticipated.

### Expected UX
- If `issuer` is set and they are creating/editing the aws config, then if they have the option to toggle accessType, the issuer should always show the current globally set value, not what was editing before toggling. e.g. it's a reset back to the original globally set value.
- If a user is not editing or creating WIF related fields on aws configuration, then issuer should not show on the details page or be hit via an get on `configuration.index` route or post on `configure-aws.ts` component.
- If a user ONLY creates/updates the issuer value that value will show on the configuration details page only if there are other WIF related fields. That means it's possible for a user to update/create an issuer and then not see the value on the details page. That's an edge case not really worth solving for at this point, IMO.

https://github.com/user-attachments/assets/4d07fdde-1a12-49b8-969e-f338aeced703

### CODE REVIEW

File: ui/tests/integration/components/secret-engine/configure-aws-test.js [hunk 1]
```diff
@@ -222,107 +286,115 @@ module('Integration | Component | SecretEngine/ConfigureAws', function (hooks) {
...
       });
 
-      // ISSUER FIELD TESTING
-      // the other tests where issuer is not passed do not show modals, so we only need to test when the modal should show up
-      test('it shows existing issuer, shows modal when saving changes, and does not call APIs on cancel', async function (assert) {
-        this.issuer = 'http://foo.bar';
-        this.server.post('/identity/oidc/config', () => {
-          assert.notOk(true, 'request should not be made to issuer config endpoint');
+      module('issuer field tests', function () {
+        // the other tests where issuer is not passed do not show modals, so we only need to test when the modal should shows up
```
Comment (hashishaw) [inline ui/tests/integration/components/secret-engine/configure-aws-test.js:290]: There was an explicit test and scenario where if the issuer config was not read, we should show a modal warning them that it could override it on save. I think that scenario has been refactored out of working

File: ui/tests/integration/components/secret-engine/configure-aws-test.js [hunk 2]
```diff
@@ -123,6 +136,24 @@ module('Integration | Component | SecretEngine/ConfigureAws', function (hooks) {
...
+        assert
+          .dom(GENERAL.inputByAttr('issuer'))
+          .hasValue(this.issuerConfig.issuer, 'issuer is what is sent in my the model on first load');
+        await fillIn(GENERAL.inputByAttr('issuer'), 'http://ive-changed');
+        await click(SES.aws.accessType('iam'));
+        await click(SES.aws.accessType('wif'));
+        assert
+          .dom(GENERAL.inputByAttr('issuer'))
+          .hasValue(
+            this.issuerConfig.issuer,
```
Comment (hashishaw) [inline ui/tests/integration/components/secret-engine/configure-aws-test.js:152]: This isn't a very reliable test because we're just checking that the value is what we pass it. A better test is checking that it has the value 'issuer' which is what it was originally

**Review by hashishaw — CHANGES_REQUESTED** [review]

File: ui/app/adapters/identity/oidc/config.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        ...resp,
+        id: resp.request_id, // id required for ember data
+      };
+    });
+  }
+
+  createOrUpdate(store, type, snapshot) {
+    const serializer = store.serializerFor(type.modelName);
+    const data = serializer.serialize(snapshot);
+    if (!data.id) data.id = data.issuer;
```
Comment (hellobontempo) [inline ui/app/adapters/identity/oidc/config.js:23]: Is `id` required for a POST request? It seems like we don't need this line since the API generates an id for us.

Comment (Monkeychip) [reply ui/app/adapters/identity/oidc/config.js:23]: It's a testing thing. I can leave a comment or attack tmrw with a fresh brain why it kept failing. But it would say request needs id.

File: ui/app/components/secret-engine/configure-aws.ts [hunk 4]
```diff
@@ -111,15 +94,15 @@ export default class ConfigureAwsComponent extends Component<Args> {
     waitFor(async (event: Event) => {
       event?.preventDefault();
       this.resetErrors();
-      const { leaseConfig } = this.args;
+      const { leaseConfig, issuerConfig } = this.args;
       // Note: only aws/lease-config model has validations
       const isValid = this.validate(leaseConfig);
       if (!isValid) return;
-      if (this.issuerConfig.dirty) {
+      if (Object.keys(issuerConfig?.changedAttributes()).some((item) => item)) {
```
[the remark this answers is not in the document]
Comment (Monkeychip) [reply ui/app/components/secret-engine/configure-aws.ts:101]: When you've overthought something 🫠... thank you.

File: ui/app/adapters/identity/oidc/config.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import ApplicationAdapter from '../../application';
+
+export default class IdentityOidcConfig extends ApplicationAdapter {
+  namespace = 'v1';
+
+  queryRecord() {
+    return this.ajax(`${this.buildURL()}/identity/oidc/config`, 'GET').then((resp) => {
+      return {
+        ...resp,
+        id: resp.request_id, // id required for ember data
```
[the remark this answers is not in the document]
Comment (Monkeychip) [reply ui/app/adapters/identity/oidc/config.js:15]: Good call.

File: ui/app/components/secret-engine/configure-aws.ts [hunk 6]
```diff
@@ -80,18 +59,22 @@ export default class ConfigureAwsComponent extends Component<Args> {
...
   constructor(owner: unknown, args: Args) {
     super(owner, args);
-
-    // the following checks are only relevant to enterprise users and those editing an existing root configuration.
+    // the following checks are only relevant to enterprise users
+    if (this.version.isCommunity) return;
+    // keep track of the original issuer so that when a user toggles the accessType we can return to the original value
+    this.originalIssuer = this.args.issuerConfig.issuer || '';
+    // the following checks are only relevant to those editing an existing root configuration
+    if (this.args.rootConfig.isNew) return;
```
[the remark this answers is not in the document]
Comment (Monkeychip) [reply ui/app/components/secret-engine/configure-aws.ts:74]: Welp, I meant to delete line 75, the two now break it up.

File: ui/tests/integration/components/secret-engine/configure-aws-test.js (hunk 2 above)
Comment (Monkeychip) [reply ui/tests/integration/components/secret-engine/configure-aws-test.js:152]: A little confusing here, and maybe I can clear this up somehow in the test. But the value passed is the global value; what is on the issuerConfig. The test is changing this value then toggling to IAM values and then toggling back and just making sure that the changed value doesn't stay. So we do want to check what's on the issuerConfig. 

Screenshots:
1. on load. 
image
2. fillIn (typing something in to simulate a user thinking about filling in something but then changing their mind).
image
3. Toggle to iam and then back and it's still the original passed in value
image

File: ui/app/components/secret-engine/configure-aws.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Monkeychip) [reply ui/app/components/secret-engine/configure-aws.ts:95]: Welp my test all fail if I remove the some check. It looks like in the tests it returns true for hasDirtyAttributes even if I've done nothing but load the form. I'll try and problem solve tmrw.

File: ui/tests/integration/components/secret-engine/configure-aws-test.js (hunk 1 above)
Comment (Monkeychip) [reply ui/tests/integration/components/secret-engine/configure-aws-test.js:290]: Okay, I think I see where i went wrong there. I added in two more test that specifically check the modal message per the capabilities check of canRead. Let's chat tmrw however, if we want any failure on the identity/oidc/config endpoint to result in canRead === false.

File: ui/tests/integration/components/secret-engine/configure-aws-test.js [hunk 7]
```diff
@@ -353,6 +425,28 @@ module('Integration | Component | SecretEngine/ConfigureAws', function (hooks) {
...
       });
     });
+    test('it does not send issuer on save', async function (assert) {
+      assert.expect(4);
+      await this.renderComponent();
+      this.server.post(configUrl('aws', this.id), () => {
+        assert.true(true, 'post request was made to config/root. test should pass.');
+      });
+      this.server.post('/identity/oidc/config', () => {
+        assert.true(false, 'post request was made to update issuer. test should fail.');
```
[the remark this answers is not in the document]
Comment (Monkeychip) [reply ui/tests/integration/components/secret-engine/configure-aws-test.js:461]: oh great idea!

File: ui/tests/integration/components/secret-engine/configure-aws-test.js (hunk 1 above)
Comment (hashishaw) [reply ui/tests/integration/components/secret-engine/configure-aws-test.js:290]: We didn't have a failure on read before because generally it doesn't matter, unless they update it we want to warn them

Comment (Monkeychip) [reply ui/tests/integration/components/secret-engine/configure-aws-test.js:292]: Reverted back to previous behavior where any failure we show this specific warning message "if it exists".

**Review by hashishaw — APPROVED** [review]
Excellent improvements, thank you!
